### PR TITLE
[WM-1951] Struct builder defaults inputs to none

### PR DESCRIPTION
**Pull request by dillydally414**
### What
- Struct builder default field value is `{ type: "none" }` 
- StructBuilder -> typescript

### Why
- Optional inputs in a struct should have a default source of none

### Testing strategy
- Validated against workflow with inner structs

### CODE REVIEW

**Comment by cjllanwarne** [comment]
As in https://github.com/DataBiosphere/terra-ui/pull/4526, could you point out which are the active code changes vs background changes for the typescript migration?

File: src/workflows-app/utils/submission-utils.ts [hunk 1]
```diff
@@ -102,9 +103,11 @@ export const inputSourceLabels = {
 export const inputSourceTypes = _.invert(inputSourceLabels);
 
 export const inputTypeParamDefaults = {
-  literal: { parameter_value: '' },
-  record_lookup: { record_attribute: '' },
-  object_builder: { fields: [] },
+  literal: () => ({ parameter_value: '' }),
+  record_lookup: () => ({ record_attribute: '' }),
+  object_builder: (inputType: StructInputType) => ({
+    fields: inputType.fields.map((field) => ({ name: field.field_name, source: { type: 'none' } })),
```
Comment (dillydally414) [inline src/workflows-app/utils/submission-utils.ts:109]: Primary change pt. 1

File: src/workflows-app/components/inputs-common.js [hunk 2]
```diff
@@ -204,7 +204,7 @@ export const InputSourceSelect = (props) => {
           type: newType,
         };
       } else {
-        const paramDefault = _.get(newType, inputTypeParamDefaults);
+        const paramDefault = inputTypeParamDefaults[newType](inputType);
```
Comment (dillydally414) [inline src/workflows-app/components/inputs-common.js:179]: Primary change pt 2.

**Review by dillydally414 — COMMENTED** [review]
Apologies for not separating these out better!

**Review by cjllanwarne — APPROVED** [review]

**Review by kpierre13 — APPROVED** [review]

**Pull request merged**